05/app.js

```diff
@@ -0,0 +1,59 @@
+function Student(firstName, lastName) {
+	this.firstName = firstName;
+	this.lastName = lastName;
+	this.grades = {};
+}
+
+Student.prototype.addGrade = function (subjectName, grade) {
+	if (typeof this.grades[subjectName] === "undefined") {
+		this.grades[subjectName] = [];
+	}
+	this.grades[subjectName].push(grade);
+};
+
+Student.prototype.getAverageGrade = function (subject) {
+	if (typeof subject === "undefined") {
+		// Średnia ze wszystkich ocen;
+		let array = [];
+		for (const key in this.grades) {
+			let allGrades = this.grades[key];
+			allGrades.forEach((el) => array.push(el));
+		}
+    const length = array.length;
+    const sum = array.reduce(function (acc, curr) {
+			return acc + curr;
+		}, 0);
+		let avgAllGrades = sum / length;  
+		console.log(avgAllGrades);
```

Comment:
Tutaj powinno być `return`.

Traktuj `console.log` jako coś co wykorzystujesz dla siebie (jako informacja zwrotna podczas pisania kodu).
Docelowo `return` będzie elementem, który zwraca informacje z funkcji i to jego będziesz używać w 99.9%.